cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NotificationFeedHttpHandler.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.gateway.auth.Authenticator;
+import co.cask.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import co.cask.cdap.notifications.feeds.NotificationFeed;
+import co.cask.cdap.notifications.feeds.NotificationFeedException;
+import co.cask.cdap.notifications.feeds.NotificationFeedManager;
+import co.cask.cdap.notifications.feeds.NotificationFeedNotFoundException;
+import co.cask.http.HttpResponder;
+import com.google.gson.JsonSyntaxException;
+import com.google.inject.Inject;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Http handler to access the service managing {@link NotificationFeed} objects.
+ * These endpoints are only reachable internally.
+ */
+@Path(Constants.Gateway.API_VERSION_3)
+public class NotificationFeedHttpHandler extends AbstractAppFabricHttpHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(NotificationFeedHttpHandler.class);
+  private final NotificationFeedManager feedManager;
+
+  @Inject
+  public NotificationFeedHttpHandler(Authenticator authenticator, NotificationFeedManager feedManager) {
+    super(authenticator);
+    this.feedManager = feedManager;
+  }
+
+  @PUT
+  @Path("/feeds/{id}")
+  public void createFeed(HttpRequest request, HttpResponder responder, @PathParam("id") String id) {
+    try {
+      NotificationFeed combinedFeed;
+      try {
+        NotificationFeed feed = parseBody(request, NotificationFeed.class);
+        combinedFeed = new NotificationFeed.Builder(NotificationFeed.fromId(id))
+          .setDescription(feed == null ? null : feed.getDescription())
+          .build();
+      } catch (IllegalArgumentException e) {
+        throw new NotificationFeedException(e);
```

Comment:
I think it's better to response with `BAD_REQUEST` here and return, in order to distinguish between user error vs system error (responds with INTERNAL_SERVER_ERROR and also log the error).